src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql

```diff
@@ -23,10 +23,20 @@ RETURNS boolean AS $$
         FROM get_all_active_transactions() WHERE process_id = pBlockedPid;
     END IF;
 
-    RETURN EXISTS (
-      SELECT 1 FROM citus_internal_global_blocked_processes()
-        WHERE waiting_global_pid = mBlockedGlobalPid
-    );
+    IF current_setting('citus.isolation_test_check_all_blocks') = 'off' THEN
```

Comment:
the `else` part seems like superset of the `if` part. 

Why do we still need `else` some cases? If you can share a simple example where the behavior changes, I want to debug a little further to understand what is going on.


Overall, this new filter makes sense btw.

Reply:
There were 4 spec files that failed with the new block detection query. I haven't investigated the reason for these failures. All of these I think might be either bugs in gpid propagation, or bugs in the new blocking query:
1. isolation_acquire_distributed_locks.spec, which used run_commands_on_session_level_connection_to_node
2. isolation_get_distributed_wait_queries_mx.spec, which used run_commands_on_session_level_connection_to_node
3. isolation_create_distributed_table_concurrently.spec, does stuff with logical replication
4. isolation_replicate_reference_tables_to_coordinator.spec, 


I think it makes sense to investigate these further and why the new query is not good enough for them. But I also didn't want to block this PR that works for almost all tests on this investigation.